contracts/secret-storage/SecretStoreManager.sol

```diff
@@ -0,0 +1,293 @@
+     * @param _secretStoreAddress The secret store contract address.
+     */
+    /// @custom:oz-upgrades-unsafe-allow constructor
+    constructor(
+        address _secretStoreAddress
+    ) {
+        _disableInitializers();
+
+        if (address(_secretStoreAddress) == address(0)) revert SecretStoreManagerZeroAddressSecretStore();
+        
+        SECRET_STORE = ISecretStore(_secretStoreAddress);
+    }
+
+    //-------------------------------- Overrides start --------------------------------//
+
+    /// @inheritdoc ERC165Upgradeable
+    function supportsInterface(
+        bytes4 interfaceId
+    ) public view virtual override(ERC165Upgradeable, AccessControlUpgradeable) returns (bool) {
+        return super.supportsInterface(interfaceId);
+    }
+
+    /// @inheritdoc UUPSUpgradeable
+    function _authorizeUpgrade(address /*account*/) internal view override onlyRole(DEFAULT_ADMIN_ROLE) {}
+
+    //-------------------------------- Overrides end --------------------------------//
+
+    //-------------------------------- Initializer start --------------------------------//
+
+    /// @notice Thrown when the provided admin address is zero.
+    error SecretStoreManagerZeroAddressAdmin();
+
+    /**
+     * @dev Initializes the contract with the given admin.
+     * @param _admin The address of the admin.
+     */
+    function initialize(address _admin) public initializer {
+        if (_admin == address(0)) revert SecretStoreManagerZeroAddressAdmin();
+
+        __Context_init_unchained();
+        __ERC165_init_unchained();
+        __AccessControl_init_unchained();
+        __UUPSUpgradeable_init_unchained();
+
+        _grantRole(DEFAULT_ADMIN_ROLE, _admin);
+    }
+
+    //-------------------------------- Initializer end --------------------------------//
+
+    bytes32 public constant SECRET_MANAGER_ROLE = keccak256("SECRET_MANAGER_ROLE");
+
+    /// @custom:oz-upgrades-unsafe-allow state-variable-immutable
+    ISecretStore public immutable SECRET_STORE;
+
+    //-------------------------------- SecretStore start --------------------------------//
+
+    struct SecretStorage {
+        uint256 storageCapacity;
+        uint256 storageOccupied;
+        uint256 lastAliveTimestamp;
+        uint256 deadTimestamp;
+        uint256[] ackSecretIds;
+    }
+
+    // enclaveAddress => Storage node details
+    mapping(address => SecretStorage) public secretStorage;
+
+    error SecretStoreManagerInvalidSecretStore();
+
+    //----------------------------------- SecretStore end -----------------------------------------//
+
+    //----------------------------- SecretManagerRole functions start --------------------------------//
+
+    //-------------------------------- internal functions start ----------------------------------//
+
+    function _register(
+        address _enclaveAddress,
+        uint256 _storageCapacity
+    ) internal {
+        if(_msgSender() != address(SECRET_STORE))
```

Comment:
use modifier, check should happen inside external function. do not use _msgSender here